pkg/cloud/azure/services/virtualmachines/virtualmachines_test.go

```diff
@@ -38,3 +44,112 @@ func TestGetTagListFromSpec(t *testing.T) {
 		}
 	}
 }
+
+func TestDeriveVirtualMachineParameters(t *testing.T) {
+	testCases := []struct {
+		name       string
+		updateSpec func(*Spec)
+		validate   func(*WithT, *compute.VirtualMachine)
+	}{
+		{
+			name:       "Unspecified security profile",
+			updateSpec: nil,
+			validate: func(g *WithT, vm *compute.VirtualMachine) {
+				g.Expect(vm.SecurityProfile).To(BeNil())
+			},
+		},
+		{
+			name: "Security profile with EncryptionAtHost set to true",
+			updateSpec: func(vmSpec *Spec) {
+				vmSpec.SecurityProfile = &v1beta1.SecurityProfile{EncryptionAtHost: to.BoolPtr(true)}
+			},
+			validate: func(g *WithT, vm *compute.VirtualMachine) {
+				g.Expect(vm.SecurityProfile).ToNot(BeNil())
+				g.Expect(vm.SecurityProfile.EncryptionAtHost).ToNot(BeNil())
+				g.Expect(*vm.SecurityProfile.EncryptionAtHost).To(BeTrue())
+			},
+		},
+		{
+			name: "Security profile with EncryptionAtHost set to false",
+			updateSpec: func(vmSpec *Spec) {
+				vmSpec.SecurityProfile = &v1beta1.SecurityProfile{EncryptionAtHost: to.BoolPtr(false)}
+			},
+			validate: func(g *WithT, vm *compute.VirtualMachine) {
+				g.Expect(vm.SecurityProfile).ToNot(BeNil())
+				g.Expect(vm.SecurityProfile.EncryptionAtHost).ToNot(BeNil())
+				g.Expect(*vm.SecurityProfile.EncryptionAtHost).To(BeFalse())
+			},
+		},
+		{
+			name: "Security profile with EncryptionAtHost unset",
+			updateSpec: func(vmSpec *Spec) {
+				vmSpec.SecurityProfile = &v1beta1.SecurityProfile{EncryptionAtHost: nil}
+			},
+			validate: func(g *WithT, vm *compute.VirtualMachine) {
+				g.Expect(vm.SecurityProfile).ToNot(BeNil())
+				g.Expect(vm.SecurityProfile.EncryptionAtHost).To(BeNil())
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
```

Comment:
I don't think we have a loop capturing issue here, this isn't normally needed since the range variable is a reference not a pointer. You can test this by breaking one of the test cases and removing this line, only the one should fail, not all of them